### ISSUE TITLE
Trucks dumping materials in locations I've marked for them not to

### ISSUE THREAD

**Issue by NextChapters** [MaFi-Games/Captain-of-Industry-issues#1430]
### Issue description

I've got a few zones and mining towers on my map to manage where mining materials get dumped - like "dispose of dirt next to the brick factory" or "put garbage in the ocean."  But I've noticed that trucks don't pay attention to the restrictions I've set - they're dumping dirt in the ocean, where the tower specifically says only to throw garbage there.  

Not sure if it's a proximity issue, they're picking the closest location that has available space for dumping and that's overwriting the defined area that's further away.  

![image](https://github.com/MaFi-Games/Captain-of-Industry-issues/assets/142436688/cd9caf1a-bf50-45e2-92ed-b677e4fbff3f)

Here's an example.  I'm digging up dirt from the pile highlighted in yellow, and the trucks are dumping it in the ocean next to it.  I've marked this zone to only accept waste.

### Reproduction steps

I've got zones designated not to receive products but they still do.

### Game version and build number

v0.6.1 (b323)

### Operating system

Windows

**Comment by poltime** [comment]
Waste will be dumped only within the yellow rectangle.

**Comment by NextChapters** [comment]
Right, that's what I was expecting.  My trucks were dumping dirt in the yellow rectangle though, even though the tower didn't have it marked to receive and there's another tower decently close that allows for dumping dirt and has space.

**Comment by MaFi-Games** [comment]
Can you please attach your save file so we can investigate?
